titus-common/src/main/java/com/netflix/titus/common/framework/simplereconciler/ReconcilerActionProviderPolicy.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.common.framework.simplereconciler;
+
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * {@link ReconcilerActionProviderPolicy} provides evaluation rules and ordering. Reconciler actions come from two
+ * main sources: external (user actions), and internal. Only actions coming from one source can be processed at a time.
+ */
+public class ReconcilerActionProviderPolicy {
+
+    /**
+     * Priority assigned to the external change actions. Internal action providers with a lower priority value will be always
+     * executed before this one.
+     */
+    public static final int DEFAULT_EXTERNAL_ACTIONS_PRIORITY = 100;
+
+    /**
+     * Default policy assigned to the external change action provider.
+     */
+    private static final ReconcilerActionProviderPolicy DEFAULT_EXTERNAL_ACTION_PROVIDER_POLICY = newBuilder()
+            .withPriority(DEFAULT_EXTERNAL_ACTIONS_PRIORITY)
+            .withExecutionInterval(Duration.ZERO)
+            .build();
+
+    /**
+     * Execution priority. The lower the value, the higher the priority.
+     */
+    private final int priority;
+
+    /**
+     * Evaluation frequency. It is guaranteed the time between a subsequent provider evaluations is at least the
```

Comment:
I'm not sure I follow exactly what the `executionInterval`  is supposed to be doing. Is it saying there should be a delay of `executionInterval` between evaluations for a specific actions provider so that it doesn't create too many actions? If so, it seems like this would potentially create delay in cases where there isn't pressure and we could process these actions faster.

Reply:
The primary motivation is that sometimes there are evaluation rules that are expensive to compute and running them too tightly is too expensive. One is example is a full reconciliation cycle that should be run every so often (let's say 60sec). With this API you would configure two action providers: one for slow reconciliation cycle and one for the regular processing with no delay.